cmd/scollector/collectors/interval.go

```diff
@@ -48,10 +49,16 @@ func (c *IntervalCollector) Run(dpchan chan<- *opentsdb.DataPoint) {
 		}
 		next := time.After(interval)
 		if c.Enabled() {
+			timeStart := time.Now()
 			md, err := c.F()
+			timeFinish := time.Since(timeStart)
+			result := 0
 			if err != nil {
 				slog.Errorf("%v: %v", c.Name(), err)
+				result = 1
 			}
+			Add(&md, "scollector.collector.duration", timeFinish.Seconds(), opentsdb.TagSet{"collector": c.Name()}, metadata.Gauge, metadata.Second, "Duration in seconds for each collector run.")
```

Comment:
Since the tags are the same here, we can set `tags := opentsdb.TagSet{"collector": c.Name()}` above the for loop on line 45.